resources/event-sources/templates/100-pre-upgrade-job.yaml

```diff
@@ -0,0 +1,136 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+rules:
+  - apiGroups: ["extensions"]
+    resources: ["deployments"]
+    verbs: ["get", "delete"]
+  - apiGroups: [""]
+    resources: ["services"]
+    verbs: ["get", "delete"]
+  - apiGroups: [""]
+    resources: ["serviceaccounts"]
+    verbs: ["get", "delete"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: event-sources-pre-upgrade
+  namespace: kyma-system
+  annotations:
+    helm.sh/hook: pre-upgrade
+    helm.sh/hook-weight: "0"
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: event-sources-pre-upgrade
+subjects:
+  - kind: ServiceAccount
+    name: event-sources-pre-upgrade
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: event-sources-pre-upgrade-helm
+rules:
+  - apiGroups: [""]
+    resources: ["pods", "pods/status"]
+    verbs: ["list", "get"]
+  - apiGroups: [""]
+    resources: ["pods/portforward"]
+    verbs: ["create", "get", "delete"]
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: ["get"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: event-sources-pre-upgrade-helm
+subjects:
+  - kind: ServiceAccount
+    name: event-sources-pre-upgrade
+    namespace: kyma-system
+roleRef:
+  kind: ClusterRole
+  name: event-sources-pre-upgrade-helm
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: event-sources-pre-upgrade-job
+  namespace: kyma-system
+  annotations:
```

Comment:
Let's add the following annotations.
```
annotations:
  "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
```

Reply:
before-hook-creation is default.

Reply:
It is not the default value with helm 2.x. More details are [here](https://v2.helm.sh/docs/charts_hooks/#automatically-delete-hook-from-previous-release)ö. Sadly we don't use helm 3.x where it is default.

Reply:
OK, done.